cmd/cli/evaluate.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/thomaspoignant/go-feature-flag/ffuser"
+	"github.com/thomaspoignant/go-feature-flag/retriever"
+	"github.com/thomaspoignant/go-feature-flag/variation"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+type EvaluationResult struct {
+	TrackEvents   bool        `json:"trackEvents"`
+	VariationType string      `json:"variationType"`
+	Failed        bool        `json:"failed"`
+	Version       string      `json:"version"`
+	Reason        string      `json:"reason"`
+	ErrorCode     string      `json:"errorCode"`
+	Value         interface{} `json:"value"`
+	Cacheable     bool       `json:"cacheable"`
+}
+
+type FlagConfig struct {
+	Flags map[string]*variation.Flag `yaml:"flags"`
+}
+
+var (
+	configFile        string
+	flagName         string
+	evaluationContext string
+)
+
+func init() {
+	evaluateCmd := &cobra.Command{
+		Use:   "evaluate",
+		Short: "Evaluate feature flags",
+		Long:  `Evaluate feature flags based on configuration and context`,
+		RunE:  runEvaluate,
+	}
+
+	evaluateCmd.Flags().StringVarP(&configFile, "config", "c", "", "Path to flag configuration file (required)")
+	evaluateCmd.Flags().StringVarP(&flagName, "flag", "f", "", "Name of the flag to evaluate")
+	evaluateCmd.Flags().StringVarP(&evaluationContext, "evaluation-context", "e", "{}", "Evaluation context in JSON format")
+
+	evaluateCmd.MarkFlagRequired("config")
+
+	rootCmd.AddCommand(evaluateCmd)
+}
+
+func runEvaluate(cmd *cobra.Command, args []string) error {
+	config, err := loadConfig(configFile)
+	if err != nil {
+		return err
+	}
+
+	context, err := parseContext(evaluationContext)
+	if err != nil {
+		return err
+	}
+
+	if flagName == "" {
+		return evaluateAllFlags(config, context)
+	}
+
+	return evaluateFlag(flagName, config, context)
+}
+
+func loadConfig(path string) (*FlagConfig, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading config file: %w", err)
+	}
+
+	var config FlagConfig
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("error parsing config file: %w", err)
+	}
+
+	return &config, nil
+}
+
+func parseContext(contextJSON string) (*ffuser.User, error) {
+	var context map[string]interface{}
+	if err := json.Unmarshal([]byte(contextJSON), &context); err != nil {
+		return nil, fmt.Errorf("error parsing evaluation context: %w", err)
+	}
+
+	targetingKey, ok := context["targetingKey"].(string)
+	if !ok {
+		return nil, fmt.Errorf("targetingKey must be a string")
+	}
+
+	user := ffuser.NewUser(targetingKey)
+	for k, v := range context {
+		if k != "targetingKey" {
+			user.Custom[k] = v
```

Comment:
**issue:** `Custom` is not a public attribute so you can't add things in the context like this.

**suggestion:** Use the function `AddCustomAttribute(k, v)` instead
```suggestion
			evalCtx.AddCustomAttribute(k, v)
```